src/pg/decoder.cr

```diff
@@ -54,22 +54,59 @@ module PG
     struct StringDecoder
       include Decoder
 
+      UUID_OID = 2950
+
       def_oids [
-        19,   # name (internal type)
-        25,   # text
-        142,  # xml
-        705,  # unknown
-        1042, # blchar
-        1043, # varchar
+        19,       # name (internal type)
+        25,       # text
+        142,      # xml
+        705,      # unknown
+        1042,     # blchar
+        1043,     # varchar
+        UUID_OID, # uuid
       ]
 
-      def decode(io, bytesize)
+      def decode(io, bytesize, oid)
+        if oid == UUID_OID
```

Comment:
Well, the initial way I approached this was to keep the decoders as they were. For example for Time we would still have `TimeDecoder` and `DateDecoder`. The problem I had with that is for decoding arrays there's a `{% for %w(Bool Int16 Int32 ...) %}` macro that defines which decoder to use for a given type. For `Int32` it's `Int32Decoder` and so on. But for `Time` it could be `TimeDecoder` or `DateDecoder`. What I did then was to check which decoder has the `oid` that was in the array header's oid. The code is something like `decoders.find { |decoder| decoder.oids.includes?(header.oid) }`. It also made the code more complex because this was specific for Time... though then I realized it was the same problem with `String`.

It's so much simpler if we have "`Time` is decoded with `TimeDecoder`" and that's it.

It's true that this adds a runtime check, but I believe this check won't affect performance, after all it's just an int comparison. Also, we would still have to do some kind of check when decoding arrays, and that check was going to be more expensive (the code I mentioned above).

I think this makes the code a bit more complex on one hand (having to pass the `oid`) but it also makes it simpler because there's a single decoder for each type.